server/service/appconfig.go

```diff
@@ -392,32 +392,42 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p []byte, applyOpts fle
 		}
 	}
 
-	delJira, err := fleet.ValidateJiraIntegrations(ctx, storedJiraByProjectKey, newAppConfig.Integrations.Jira)
-	if err != nil {
-		if errors.As(err, &fleet.IntegrationTestError{}) {
-			return nil, ctxerr.Wrap(ctx, &fleet.BadRequestError{
-				Message: err.Error(),
-			})
+	// NOTE: the frontend will always send all integrations back when making
+	// changes, so as soon as Jira or Zendesk has something set, it's fair to
+	// assume that integrations are being modified and we have the full set of
+	// those integrations. When deleting, it does send empty arrays (not nulls),
+	// so this is fine - e.g. when deleting the last integration it sends:
```

Comment:
looking at the code, that seems to be the case indeed. I would still wait for Gabe or Sarah to confirm

Reply:
I believe this is the case today, but I'm hesitant to say that we should always assume this will be the case. There's currently nothing to guarantee on the frontend that it always sends all integrations when modifying. I worry that even if this is the case today, a subtle change on the frontend could upset these assumptions because the client-side validations and typings are quite weak and scattered across the codebase. Ideally, the frontend would refactor to centralize this behavior and create some explicit tests to be sure the current implicit behavior doesn't change. Any thoughts on how we ought to approach this @ghernandez345?

Reply:
@gillespi314 

> I believe this is the case today, but I'm hesitant to say that we should always assume this will be the case.

I agree with you that there's a non-zero risk that this could be broken in the future. On the backend (i.e. if we break that logic directly on the backend, regardless of how the frontend sends it), this would be caught by the integrations tests added in this PR. If there's a way we could prevent such a regression on the frontend too, it would be ideal (an end-to-end test?). The break would be relatively obvious because any call with missing integrations would result with those integrations being removed.

That being said, I think this fix should still move forward as it does fix the bug, and a subsequent PR could be made for the frontend to prevent a future regression.

Reply:
That makes sense to me. Perhaps we could create a follow up ticket for frontend to ensure that we have integration coverage and/or to add this to the e2e tests being run by QAWolf?

Reply:
@gillespi314 Sounds good, thanks! I'll create a follow-up ticket with your suggestions.

Reply:
Created this ticket: https://github.com/fleetdm/fleet/issues/13756